cmd/helm/installer/install.go

```diff
@@ -129,24 +129,23 @@ func generateLabels(labels map[string]string) map[string]string {
 }
 
 // parseNodeSelectors takes a comma delimited list of key=values pairs and returns a map
-func parseNodeSelectors(labels string) map[string]string {
+func parseNodeSelectors(labels string, m map[string]string) map[string]string {
```

Comment:
- [ ] This function got strange with this change. It now mutates a supplied map in place, and returns that same map, though the signature can't promise that the returned map is the same as the map supplied as an argument. A caller might wonder whether he needs the return value. A caller might wonder whether his input argument is safe against mutation.  

  What if you had `parseNodeSelectors` create and return a fresh map, eliminating this confusion, and then add your "os=linux" selector into the map returned from the call in `generateDeployment`?